### PR TITLE
Fix LT-18781:Crash appears on "Lexicon Edit" pane when changing category

### PR DESCRIPTION
**Pull request by vkarthim**
*Added one more condition to set null
*InflFeatsOA also be null when invalid features.
*Added test case OnRefresh_SetNullInflFeatsOAWhenInvalidFeature

---
This change is [Reviewable](https://reviewable.io/reviews/sillsdev/liblcm/27)

### CODE REVIEW

File: tests/SIL.LCModel.Tests/DomainImpl/LingTests.cs [hunk 1]
```diff
@@ -2064,6 +2065,71 @@ public void MoStemMsa_EqualMsa()
...
+			doc.LoadXml(MoreCellarTests.ksFS1);
+			XmlNode itemNeut = doc.SelectSingleNode("/item/item[3]");
+			XmlNode itemFem = doc.SelectSingleNode("/item/item[2]");
+			var featStruc1 = Cache.ServiceLocator.GetInstance<IFsFeatStrucFactory>().Create();
+			var featStruc2 = Cache.ServiceLocator.GetInstance<IFsFeatStrucFactory>().Create();
+			infAffixMsa1.InflFeatsOA = featStruc1;
+			infAffixMsa2.InflFeatsOA = featStruc2;
+			infAffixMsa1.InflFeatsOA.AddFeatureFromXml(itemNeut, Cache.LangProject.MsFeatureSystemOA);
+			infAffixMsa2.InflFeatsOA.AddFeatureFromXml(itemFem, Cache.LangProject.MsFeatureSystemOA);
+
```
Comment (jasonleenaylor) [inline tests/SIL.LCModel.Tests/DomainImpl/LingTests.cs]: Add an assert here to verify that infAffixMsa1.InflFeatsOA is not null before we change the POS.

File: tests/SIL.LCModel.Tests/DomainImpl/LingTests.cs [hunk 2]
```diff
@@ -2064,6 +2065,71 @@ public void MoStemMsa_EqualMsa()
...
+			// reset POS
```
Comment (jasonleenaylor) [inline tests/SIL.LCModel.Tests/DomainImpl/LingTests.cs]: Change comment to // change to POS with incompatible features

File: tests/SIL.LCModel.Tests/DomainImpl/LingTests.cs (hunk 1 above)
Comment (vkarthim) [reply tests/SIL.LCModel.Tests/DomainImpl/LingTests.cs]: Ok

File: tests/SIL.LCModel.Tests/DomainImpl/LingTests.cs (hunk 2 above)
Comment (vkarthim) [reply tests/SIL.LCModel.Tests/DomainImpl/LingTests.cs]: Ok

**Pull request merged**